src/TextInput/TextInput.tsx

```diff
@@ -68,8 +70,22 @@ export class TextInput extends React.Component<Props, State> {
   static propTypes = propTypes;
   static defaultProps = defaultProps;
 
+  static validateProps(props) {
+    if (props.required && props.optional) {
+      throw new Error("You cannot pass both `required` and `optional` on a TextArea.");
+    }
+
+    if (["readOnly", "disabled", "inFocus"].filter(x => props[x]).length > 1) {
```

Comment:
`inFocus` is actually a `state` value and not a `prop`. You'll see this if you look at the `Props` and `State` type definitions for the component:

https://github.com/Clever/components/blob/59a72dd0f241f073bc7c403170e0b3bb651a3685/src/TextInput/TextInput.tsx#L13-L39

So no need to include `"inFocus"` in this array right here. (I see that you copied this logic from `<TextArea>`. We may wanna update that validation method as well.)

-----

In terms of backwards compatibility, I wouldn't be surprised if existing consumers accidentally supply both a `readOnly` and `disabled` prop to some `<TextInput>` components. And since this code will generate a run-time error (as opposed to a compile-time error), I could see one of our client-side JS bundles crashing unexpectedly because of this.

Rather than throwing an error, a safer alternative might be to emit a `console.warning` or `console.error`. We should update the `<TextArea>` `validateProps` method as well. The main goal here is to notify developers that they've made a mistake. But we don't want that notification method to have consequences of its own (i.e. unexpected JS crashes)

Reply:
Yea makes sense. Since there are only 2 props I just changed it to use '&&'.

Kept throw new error for 'required && optional' as that was there in textArea already and a new feature in textInput so no consumer is using it yet.